linkerd/identity/src/imp/openssl.rs

```diff
@@ -0,0 +1,345 @@
+use std::sync::Arc;
+use std::time::SystemTime;
+use std::{error, fmt};
+
+#[cfg(feature = "boring-tls")]
+use boring::{
+    error::ErrorStack,
+    pkey::{PKey, Private},
+    stack::Stack,
+    x509::{
+        store::{X509Store, X509StoreBuilder},
+        {X509StoreContext, X509VerifyResult, X509},
+    },
+};
+#[cfg(not(feature = "boring-tls"))]
+use openssl::{
+    error::ErrorStack,
+    pkey::{PKey, Private},
+    stack::Stack,
+    x509::{
+        store::{X509Store, X509StoreBuilder},
+        {X509StoreContext, X509VerifyResult, X509},
+    },
+};
+
+use tracing::{debug, warn};
+
+use crate::{LocalId, Name};
+use std::fmt::Formatter;
+
+#[derive(Clone, Debug)]
+pub struct Key(pub Arc<PKey<Private>>);
```

Comment:
Yeah I totally agree.
I'm looking into what is the best solution here. The reason why this is public at this time is because of the identity vs tls modules.

When constructing a TlsConnector I need access to the primitive type Pkey<Private> from within the tls module.
I'm pretty sure this is something that can be solved by implementing `From` and `Into` traits but I'm not seeing the solution at this time.